gradle/verification-metadata.xml

```diff
@@ -784,6 +784,11 @@
             <sha256 value="7664cf8c6f2adadf600287812b32878277beda54912eab9d4c2932cd50cb704a" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.nimbusds" name="oauth2-oidc-sdk" version="10.7.1">
```

Comment:
you may already be planning this (so sorry if this is premature) ...when adding new versions, please remove the old one just to keep this file small-ish. The tooling only adds and removal requires manual removal.

Reply:
Thanks for the pointer! Not premature at all, will remove the old entries once I've untangled the breaking changes around the version bump for `oauth2-oidc-sdk`.

Reply:
Hmm looks like the dependencies are getting cached (?), resulting in build failure when I remove them:

https://gradle-enterprise.elastic.co/s/pc7ol75urjudu/console-log?page=6#L5148

```
* What went wrong:	
Execution failed for task ':test:fixtures:hdfs3-fixture:compileJava'.	
> Dependency verification failed for configuration ':test:fixtures:hdfs3-fixture:compileClasspath':	
    - On artifact json-smart-2.3.jar (net.minidev:json-smart:2.3) in repository 'MavenRepo2': checksum is missing from verification metadata.	
    - On artifact json-smart-2.4.2.jar (net.minidev:json-smart:2.4.2) in repository 'MavenRepo2': checksum is missing from verification metadata.
```

Checking it out but may opt to get them in, in the interest of making FF 👀

Reply:
Pretty confused here 🤔 

Running `./gradlew --write-verification-metadata sha256 help` correctly adds the new entry for `net.minidev:json-smart:2.4.10` to `gradle/verification-metadata.xml`. However, when I remove the old entries, and re-run `./gradlew --write-verification-metadata sha256 help`, they get added back. I'm not sure what is triggering this, since `./gradlew x-pack:plugin:security:dependencies | grep net.minidev:json-smart` only gives me:

```
+--- net.minidev:json-smart:2.4.10 (n)
+--- net.minidev:json-smart:2.4.10
+--- net.minidev:json-smart:2.4.10
+--- net.minidev:json-smart:2.4.10
+--- net.minidev:json-smart:2.4.10
+--- net.minidev:json-smart:2.4.10
+--- net.minidev:json-smart:2.4.10
+--- net.minidev:json-smart:2.4.10
+--- net.minidev:json-smart:2.4.10
+--- net.minidev:json-smart:2.4.10
+--- net.minidev:json-smart:2.4.10
```

I.e., I'm not seeing any dependency on the old versions. I've manually nuked `~/.gradle/caches` just in case. 

Running `./gradlew --write-verification-metadata sha256 help --debug` does produce debug lines that indicate that we're trying to resolve the old packages:

```
2023-04-11T18:26:32.985+0200 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.RepositoryChainComponentMetaDataResolver] Attempting to resolve component for net.minidev:json-smart:2.4.2 using repositories [MavenRepo, MavenRepo2, lucene-snapshots, jdk_repo_openjdk_20+36@bdc68b4b9cbc4ebcb30745c85038d91d]
2023-04-11T18:26:32.985+0200 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.CachingModuleComponentRepository] Using cached module metadata for module 'net.minidev:json-smart:2.4.2' in 'MavenRepo'
2023-04-11T18:26:32.985+0200 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.graph.builder.DependencyGraphBuilder] Visiting configuration org.elasticsearch:elasticsearch-grok:8.8.0-SNAPSHOT(apiElements)
```

But I don't get _why_.

@jakelandis do you have a gradle tips for tracking down why these are getting added back? 

What makes this more mysterious is that a [previous build](https://gradle-enterprise.elastic.co/s/vgqgn5pmjqpw4) with the old deps removed was failing, but a different build for a new PR: https://github.com/elastic/elasticsearch/pull/95162 without the old deps passes. I'm wondering if there is some mysterious caching going on somewhere that might actually be safe to ignore...

Reply:
These are getting in through test dependencies. To this end, I'm leaving the "old" versions in the verification file. I've validated that `localDistro` only includes the jar with the correct, updated version.